data_structures/binarysearchtree/binarysearchtree.go

```diff
@@ -47,66 +48,66 @@ func (b *BinarySearchTree) Insert(data int) *BinarySearchTree {
 func (b *BinarySearchTree) BFSTraversal() []int {
 	visited := []int{}
 
-	if b.root == nil {
+	if b.Root == nil {
 		return visited
 	}
 
-	consideredQueue := []leaf{}
-	node := *b.root
+	consideredQueue := []Leaf{}
+	node := *b.Root
 	consideredQueue = append(consideredQueue, node)
 	for len(consideredQueue) > 0 {
 		node = consideredQueue[0]
 		consideredQueue = consideredQueue[1:]
-		visited = append(visited, node.data)
-		if node.left != nil {
-			consideredQueue = append(consideredQueue, *node.left)
+		visited = append(visited, node.Data)
+		if node.Left != nil {
+			consideredQueue = append(consideredQueue, *node.Left)
 		}
-		if node.right != nil {
-			consideredQueue = append(consideredQueue, *node.right)
+		if node.Right != nil {
+			consideredQueue = append(consideredQueue, *node.Right)
```

Comment:
This module is required in the data_structures/main as well... please make sure that the changes are getting propagated there as well